acir/src/circuit/mod.rs

```diff
@@ -22,6 +22,11 @@ pub struct Circuit {
     pub current_witness_index: u32,
     pub opcodes: Vec<Opcode>,
     pub public_inputs: PublicInputs,
+    // Public outputs are a subset of public inputs.
+    // They describe at a high level, what witness indices
+    // should be seen as the output of executing the
+    // function that this Circuit represents
+    pub public_outputs: PublicInputs,
```

Comment:
```suggestion
   // ACIR distinguishes between the public inputs which are provided
   // externally (`public_parameters`) or calculated within the circuit (`return_values`).
   // The elements of these sets may not be mutually exclusive, i.e. a parameter may be returned from the circuit.
   // All public inputs must be provided to the verifier at verification time.
   
    /// The set of public inputs provided by the prover.
    pub public_parameters: PublicInputs,
    /// The set of public inputs calculated within the circuit.
    pub return_values: PublicInputs,
```

Reply:
We'd want to add a `fn public_inputs()-> PublicInputs` function which returns the union of these two sets as well.